src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java

```diff
@@ -45,14 +49,39 @@ public List<CaseEvent> handledEvents() {
         return EVENTS;
     }
 
+    private CallbackResponse populateCaseData(CallbackParams callbackParams) {
+        var caseData = callbackParams.getCaseData();
+        CaseData.CaseDataBuilder<?, ?> updatedCaseData = caseData.toBuilder();
+        updatedCaseData.showResponseOneVOneFlag(responseOneVOneService.setUpOneVOneFlow(caseData));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.build().toMap(objectMapper))
+            .build();
+    }
+
     private CallbackResponse aboutToSubmit(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
         CaseData updatedData = caseData.toBuilder()
             .businessProcess(BusinessProcess.ready(CLAIMANT_RESPONSE_CUI))
             .build();
-        return AboutToStartOrSubmitCallbackResponse.builder()
-            .data(updatedData.toMap(objectMapper))
-            .build();
+
+        AboutToStartOrSubmitCallbackResponse.AboutToStartOrSubmitCallbackResponseBuilder response =
+            AboutToStartOrSubmitCallbackResponse.builder()
+                .data(updatedData.toMap(objectMapper));
+
+        updateClaimEndState(response, updatedData);
+
+        return response.build();
+    }
+
+    private void updateClaimEndState(AboutToStartOrSubmitCallbackResponse.AboutToStartOrSubmitCallbackResponseBuilder response, CaseData updatedData) {
+        if (updatedData.hasClaimantAgreedToFreeMediation()) {
+            response.state(CaseState.IN_MEDIATION.name());
+        } else if (updatedData.hasApplicantRejectedRepaymentPlan() && (updatedData.getRespondent1().isCompany() || updatedData.getRespondent1().isOrganisation())) {
+            response.state(CaseState.All_FINAL_ORDERS_ISSUED.name());
```

Comment:
For Manual Determination claim end state suppose to be 'PROCEEDS_IN_HERITAGE_SYSTEM' ??

Reply:
Finally yes, business decisions.